File: Services & Background Tasks Examples/CustomBroadcastDemo/app/src/main/java/com/example/undefined/custombroadcastdemo/MainActivity.java
package com.example.undefined.custombroadcastdemo;

import android.content.IntentFilter;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

  //  ExBroadcastReceiver exBroadcastReceiver=new ExBroadcastReceiver();
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

/*        IntentFilter filter=new IntentFilter(" ");
        registerReceiver(exBroadcastReceiver,filter);*/
    }

/*    @Override
    protected void onDestroy() {
        super.onDestroy();
        unregisterReceiver(exBroadcastReceiver);
    }*/
}

File: Android-Architecture-Components-Example/RoomCodelab/app/src/main/java/com/example/roomcodelab/MainActivity.kt
package com.example.roomcodelab

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelProviders
import androidx.lifecycle.get
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private lateinit var wordViewModel : WordViewModel
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val recyclerView = findViewById<RecyclerView>(R.id.recyclerview)
        val adapter = WordListAdapter(this)
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(this)

        wordViewModel = ViewModelProviders.of(this).get(WordViewModel::class.java)
        wordViewModel.allWords.observe(this, Observer { words->
            words?.let { adapter.setWords(it) }
        })

        fab.setOnClickListener{
            val intent = Intent(this@MainActivity, NewWordActivity::class.java)
            startActivityForResult(intent, newWordActivityRequestCode)
        }

    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if(requestCode == newWordActivityRequestCode && resultCode == Activity.RESULT_OK){
            data?.let {
                val word = Word(it.getStringExtra(NewWordActivity.EXTRA_REPLAY))
                wordViewModel.insert(word)
            }
        }
        else{
            Toast.makeText(applicationContext,R.string.empty_not_saved, Toast.LENGTH_SHORT).show()
        }
    }

    companion object {
        const val newWordActivityRequestCode = 1
    }
}
